qemu/tests/virt_firmware_check_phys_bits.py

```diff
@@ -0,0 +1,98 @@
+import re
+from avocado.utils import process
+from virttest import virt_vm
+from virttest import env_process
+from virttest import error_context
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Test parameters host-phys-bits and host-phys-bits-limit.
+    host-phys-bits=on is default on rhel qemu-kvm builds,
+    so cover the default value in the test.
+    phys-bits to be covered in the test:
+    AMD processors have 40, 43 or 48 phys-bits.
+    Intel processors have 36, 39, 46 or 52 phys-bits.
+    In this case, the cpu will be configured with host-phys-bits=on
+    and host-phys-bits-limit=<phys-bits-to-test>.
+    The maximum number OVMF is using right now is 46.
+    For 5-level page inside OVMF, they don't have that yet.
+    The limit setting its higher on the host configuration doesn't hurt,
+    but OVMF just wouldn't use it.
+
+    1) boot a guest with invalid phys-bits, check the error messag
+       e.g. host-phys-bits-limit=-1, host-phys-bits-limit=1
+    2) boot a guest with valid phys-bits, check the phys-bits in guest
+       make sure it equals to the set value or equals to the maximum
+       number which host supporting
+    3) if phys-bits number is greater than 46, check the limitation
+       message in virt firmware log
+       limitation message: limit PhysBits to 46 (avoid 5-level paging)
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    phys_bits_grep_cmd = params["phys_bits_grep_cmd"]
+    host_phys_bits = process.getoutput(phys_bits_grep_cmd, shell=True).strip()
+    if not host_phys_bits.isdigit():
+        test.error("Failed to get host phys-bits, the actual output is '%s'"
+                   % host_phys_bits)
+    host_phys_bits_limit = params["host_phys_bits_limit"]
+    phys_bits_flags = params["phys_bits_flags"] % host_phys_bits_limit
+    params["cpu_model_flags"] += ",%s" % phys_bits_flags
+    err_msg = params.get("err_msg")
+    ignored_err_msg = params.get("ignored_err_msg")
+    try:
+        error_context.context("Start the vm with host-phys-bits-limit=%s."
+                              % host_phys_bits_limit, test.log.info)
+        params["start_vm"] = "yes"
```

Comment:
try best to avoid modifying a cartesian param at runtime
we can start vm after env.get_vm(vm_name)

Another way is to set the following in cfg (you may need to verify if it works :-D)
not_preprocess = yes
start_vm = yes

Reply:
Thanks for the information, I will update it. Many thanks.